aten/src/ATen/native/Loss.cpp

```diff
@@ -148,4 +148,36 @@ Tensor poisson_nll_loss(const Tensor& input, const Tensor& target, const bool lo
 
     return apply_loss_reduction(loss, reduction);
 }
+
+Tensor mse_loss(const Tensor& input, const Tensor& target, int64_t reduction) {
+  auto loss = input.sub(target).pow_(2);
+  return apply_loss_reduction(loss, reduction);
+}
+
+Tensor& mse_loss_out(Tensor&result, const Tensor& input, const Tensor& target, int64_t reduction) {
+  if (reduction != Reduction::None) {
+    auto loss = input.sub(target).pow_(2);
+    if (reduction == Reduction::Mean) {
+      at::mean_out(result, loss, 0);
+    } else {
+      at::sum_out(result, loss, 0);
+    }
+  } else {
+    at::sub_out(result, input, target).pow_(2);
```

Comment:
As example this is 2 separate CPU loops or CUDA kernels, one is for `sub`, another is for `pow`.